### PR TITLE
DOMA-1855 select with editable tag component added

### PR DESCRIPTION
**Pull request by 2000Ilya**
New SelectTags component with custom editable tags added. 
Tag can be changed clicking on i
tags-editing

### CODE REVIEW

File: apps/condo/domains/common/components/Table/Filters.tsx [hunk 1]
```diff
@@ -86,17 +87,30 @@ export const getSelectFilterDropdown = (selectProps: ComponentProps<typeof Selec
...
-                    value={selectedKeys}
-                    optionFilterProp={'label'}
-                    onChange={(e) => {
-                        setSelectedKeys(e)
-                        confirm({ closeDropdown: false })
-                    }}
-                    {...selectProps}
-                />
+                {selectProps.mode === 'tags' ?
+                    <SelectTags
```
Comment (akarjakin) [inline apps/condo/domains/common/components/Table/Filters.tsx:91]: I don't see any differecnce between this two <Select ...> props

File: apps/condo/domains/common/components/antd/SelectTags.tsx [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+    .ant-tag-close-icon {
+      vertical-align: 0.2px;
+    }
+`
+
+const tagRender = (props) => {
+    const { setSelectInputValue, tags, onChangeSelect, value, ...restProps } = props
+
+    const changeTagValue = () => {
```
Comment (akarjakin) [inline apps/condo/domains/common/components/antd/SelectTags.tsx:32]: useCallback

File: apps/condo/domains/common/components/antd/SelectTags.tsx [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    )
+}
+
+const SelectTags = <T extends string | number | { value: any, label: any }>(props: CustomSelectProps<T>) => {
+    const { eventName: propEventName, eventProperties = {}, onChange, value, ...restProps } = props
+    const [selectInputValue, setSelectInputValue] = useState('')
+    const { logEvent, getEventName } = useTracking()
+
+    const eventName = propEventName ? propEventName : getEventName(TrackingEventType.Select)
+    const componentProperties = { ...eventProperties }
```
Comment (akarjakin) [inline apps/condo/domains/common/components/antd/SelectTags.tsx:60]: useMemo

File: apps/condo/domains/common/components/antd/SelectTags.tsx [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+    }
+
+    return (
+        <DefaultSelect <T>
+            {...restProps}
+            onChange={onChangeCallback}
+            tagRender={(props) => {
+                return tagRender({ ...props, setSelectInputValue, tags: value, onChangeSelect: onChange })
+            }}
+            onSearch={(value) => {
```
Comment (akarjakin) [inline apps/condo/domains/common/components/antd/SelectTags.tsx:86]: useCallback

File: apps/condo/domains/common/components/antd/SelectTags.tsx [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+import isObject from 'lodash/isObject'
+import { jsx, css } from '@emotion/react'
+import { CustomSelectProps } from './Select'
+
+const tagCss = css`
+    height: 32px;
+    line-height: 32px;
+    margin-top: 6px;
+    margin-bottom: 6px;
+    background: #f5f5f5;
```
Comment (dkoviazin) [inline apps/condo/domains/common/components/antd/SelectTags.tsx:16]: Better to import colors from constants
```
import { colors } from '@condo/domains/common/constants/style'

```

File: apps/condo/domains/common/components/antd/SelectTags.tsx [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+            onChange(value, option)
+            setSelectInputValue('')
+        }
+    }
+
+    return (
+        <DefaultSelect <T>
+            {...restProps}
+            onChange={onChangeCallback}
+            tagRender={(props) => {
```
Comment (akarjakin) [inline apps/condo/domains/common/components/antd/SelectTags.tsx:83]: useCallback

File: apps/condo/domains/common/components/antd/SelectTags.tsx [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+
+const SelectTags = <T extends string | number | { value: any, label: any }>(props: CustomSelectProps<T>) => {
+    const { eventName: propEventName, eventProperties = {}, onChange, value, ...restProps } = props
+    const [selectInputValue, setSelectInputValue] = useState('')
+    const { logEvent, getEventName } = useTracking()
+
+    const eventName = propEventName ? propEventName : getEventName(TrackingEventType.Select)
+    const componentProperties = { ...eventProperties }
+
+    const onChangeCallback: CustomSelectProps<T>['onChange'] = (value, option) => {
```
Comment (akarjakin) [inline apps/condo/domains/common/components/antd/SelectTags.tsx:62]: useCallback

**Review by akarjakin — CHANGES_REQUESTED** [review]

File: apps/condo/domains/common/components/Table/Filters.tsx (hunk 1 above)
Comment (DanilovMaxim13) [reply apps/condo/domains/common/components/Table/Filters.tsx:91]: SelectTags and Select :) 
But I think attributes can be declared once and sent to both components

**Comment by 2000Ilya** [comment]
Problem solution does not satisfy ux of Select field.

**Pull request closed without merging**